Lib/fontTools/feaLib/builder.py

```diff
@@ -8,15 +8,19 @@
 import itertools
 
 
-def addOpenTypeFeatures(featurefile_path, font):
-    builder = Builder(featurefile_path, font)
+def addOpenTypeFeatures(font, featurefile_path=None, featurefile_data=None):
+    # we may have a path but no data, or data but no path
+    assert featurefile_path or featurefile_data
```

Comment:
Since one of the two is required it is better to raise an exception with some usage info massage rather than an assertion